kyo-core/shared/src/main/scala/kyo/Log.scala

```diff
@@ -2,10 +2,28 @@ package kyo
 
 import kyo.internal.LogPlatformSpecific
 
+final case class Log(unsafe: Log.Unsafe) extends AnyVal:
+    def level: Log.Level                                                                                  = unsafe.level
+    inline def trace(inline msg: => String)(using inline frame: Frame): Unit < IO                         = IO.Unsafe(unsafe.trace(msg))
+    inline def trace(inline msg: => String, inline t: => Throwable)(using inline frame: Frame): Unit < IO = IO.Unsafe(unsafe.trace(msg, t))
+    inline def debug(inline msg: => String)(using inline frame: Frame): Unit < IO                         = IO.Unsafe(unsafe.debug(msg))
+    inline def debug(inline msg: => String, inline t: => Throwable)(using inline frame: Frame): Unit < IO = IO.Unsafe(unsafe.debug(msg, t))
+    inline def info(inline msg: => String)(using inline frame: Frame): Unit < IO                          = IO.Unsafe(unsafe.info(msg))
+    inline def info(inline msg: => String, inline t: => Throwable)(using inline frame: Frame): Unit < IO  = IO.Unsafe(unsafe.info(msg, t))
+    inline def warn(inline msg: => String)(using inline frame: Frame): Unit < IO                          = IO.Unsafe(unsafe.warn(msg))
+    inline def warn(inline msg: => String, t: => Throwable)(using inline frame: Frame): Unit < IO         = IO.Unsafe(unsafe.warn(msg, t))
+    inline def error(inline msg: => String)(using inline frame: Frame): Unit < IO                         = IO.Unsafe(unsafe.error(msg))
+    inline def error(inline msg: => String, t: => Throwable)(using inline frame: Frame): Unit < IO        = IO.Unsafe(unsafe.error(msg, t))
+end Log
+
 /** Logging utility object for Kyo applications. */
 object Log extends LogPlatformSpecific:
 
-    private val local = Local.init[Unsafe](unsafe)
+    enum Level:
+        case Trace, Debug, Info, Warn, Error
+        def enabled(minLevel: Level): Boolean = this.ordinal >= minLevel.ordinal
```

Comment:
Will this have any issues if we need to add/remove levels?

Reply:
I've made it be based on an `Int` with some padding so we can make changes in the future